raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2252,9 +2247,12 @@ public Long scheduleAtomicAppend(int epoch, List<T> records) {
         return append(epoch, records, true);
     }
 
+    @SuppressWarnings("unchecked")
     private Long append(int epoch, List<T> records, boolean isAtomic) {
-        BatchAccumulator<T> accumulator = this.accumulator;
-        if (accumulator == null) {
+        BatchAccumulator<T> accumulator;
+        try {
+            accumulator =  (BatchAccumulator<T>) quorum.leaderStateOrThrow().accumulator();
```

Comment:
I think you should be able to remove this cast.

Reply:
Removing the cast causes this...
`Type Mismatch: Cannot convert from BatchAccumulator<Object> to BatchAccumulator<T> `

even though this is the signature of accumulator()
`public BatchAccumulator<T> accumulator()`